patches/picolibc.patch

```diff
@@ -32,3 +32,56 @@ index c757f7665..e262e1c21 100644
  
  	.bss (NOLOAD) : {
  		*(.sbss*)
+diff --git a/picolibc.ld.in b/picolibc.ld.in
+index c757f7665..d0b0c09c1 100644
+--- a/picolibc.ld.in
++++ b/picolibc.ld.in
+@@ -207,11 +207,12 @@ SECTIONS
+ 	 * across them.  We actually need memory allocated for tbss,
+ 	 * so we create a special segment here just to make room
+ 	 */
++	 /*
+ 	.tbss_space (NOLOAD) : {
+ 		. = ADDR(.tbss);
+ 		. = . + SIZEOF(.tbss);
+ 	} >ram AT>ram :ram
+-
++	*/
+ 	.bss (NOLOAD) : {
+ 		*(.sbss*)
+ 		*(.gnu.linkonce.sb.*)
+diff --git a/test/rounding-mode-sub.c b/test/rounding-mode-sub.c
+index 58372d9ef..8757760da 100644
+--- a/test/rounding-mode-sub.c
++++ b/test/rounding-mode-sub.c
+@@ -38,7 +38,7 @@
+ #include <fenv.h>
+ 
+ double
+-div(double a, double b)
++div__(double a, double b)
```

Comment:
Why?

Reply:
Without this patch test cannot be compiled due to duplicate symbol:
check-picolibc_armv6m_soft_nofp_build_rounding-mode.out 
cd /home/pioprz01/work_repo/build && /home/pioprz01/work_repo/build/llvm/bin/clang --config /home/pioprz01/work_repo/build/llvm/bin/armv6m_soft_nofp_semihost.cfg -T ../samples/ldscripts/armv6m_soft_nofp.ld -nostdlib -lc -lg -lm -lclang_rt.builtins -lsemihost -Oz /data_sda/pioprz01/LLVM-embedded-toolchain-for-Arm_fork/LLVM-embedded-toolchain-for-Arm/repos/picolibc/test/rounding-mode.c /data_sda/pioprz01/LLVM-embedded-toolchain-for-Arm_fork/LLVM-embedded-toolchain-for-Arm/repos/picolibc/test/rounding-mode-sub.c -o check-picolibc_armv6m_soft_nofp_build_rounding-mode.out
ld.lld: error: duplicate symbol: div
>>> defined at div.c:82 (/home/pioprz01/work_repo/build/picolibc/armv6m_soft_nofp/src/picolibc_armv6m_soft_nofp-build/../../../../../repos/picolibc/newlib/libc/stdlib/div.c:82)
>>>            libc_stdlib_div.c.o:(div) in archive /home/pioprz01/work_repo/build/llvm/bin/../lib/clang-runtimes/arm-none-eabi/armv6m_soft_nofp/lib/libc.a
>>> defined at rounding-mode-sub.c
>>>            /tmp/rounding-mode-sub-099cc0.o:(.text+0x1)
clang-16: error: ld.lld command failed with exit code 1 (use -v to see invocation)

Reply:
Thanks for explaining. Yes, there's a standard C `div` function. I think this is worth a PR in picolibc. For now I'd prefer to leave this test disabled, and not patch picolibc.